From aa6a4732e287b4fc44f73e65a01a87849e266412 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:55:03 -0400
Subject: [fs] ext4: use i_size_read in ext4_unaligned_aio()

Message-id: <1414058125-4183-150-git-send-email-lczerner@redhat.com>
Patchwork-id: 98173
O-Subject: [RHEL 7.1 PATCH v2 149/171] ext4: use i_size_read in ext4_unaligned_aio()
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Theodore Ts'o <tytso@mit.edu>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 6e6358fc3c3c862bfe9a5bc029d3f8ce43dc9765

 We haven't taken i_mutex yet, so we need to use i_size_read().

 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Cc: stable@vger.kernel.org
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/file.c b/fs/ext4/file.c
index 582ff74..6a78a9e 100644
--- a/fs/ext4/file.c
+++ b/fs/ext4/file.c
@@ -82,7 +82,7 @@ ext4_unaligned_aio(struct inode *inode, const struct iovec *iov,
  size_t count = iov_length(iov, nr_segs);
  loff_t final_size = pos + count;
 
- if (pos >= inode->i_size)
+ if (pos >= i_size_read(inode))
   return 0;
 
  if ((pos & blockmask) || (final_size & blockmask))
-- 
1.7.1